<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>950</width>
    <height>700</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>850</width>
    <height>680</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>10</number>
      </property>
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <property name="leftMargin">
       <number>6</number>
      </property>
      <property name="topMargin">
       <number>6</number>
      </property>
      <property name="rightMargin">
       <number>6</number>
      </property>
      <property name="bottomMargin">
       <number>6</number>
      </property>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="spacing">
         <number>6</number>
        </property>
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <item>
         <layout class="QVBoxLayout" name="topLayout">
          <property name="spacing">
           <number>6</number>
          </property>
          <item>
           <widget class="QFrame" name="algsListFrame">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_7">
             <item>
              <widget class="QLabel" name="algsListLabel">
               <property name="text">
                <string>Алгоритмы рисования</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="algorithmComboBox">
               <property name="minimumSize">
                <size>
                 <width>180</width>
                 <height>0</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="colorFrame">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_12">
             <item>
              <widget class="QLabel" name="colorChoiceLabel">
               <property name="text">
                <string>Цвет рисования</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="colorComboBox"/>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="FigureFrame">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <widget class="QLabel" name="figureLabel">
               <property name="text">
                <string>Фигура</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="figureComboBox"/>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QStackedWidget" name="CircleEllipseSW">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="circlePage">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>220</width>
               <height>0</height>
              </size>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QFrame" name="drawCircleFrame">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="frameShape">
                 <enum>QFrame::StyledPanel</enum>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_13">
                 <item>
                  <widget class="QLabel" name="drawCircleLabel">
                   <property name="text">
                    <string>Координаты Окружности</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <layout class="QGridLayout" name="circleInputGrid" columnminimumwidth="80,0,0">
                   <item row="1" column="0">
                    <widget class="QLabel" name="circleCenterLabel">
                     <property name="text">
                      <string>(Xc, Yc)</string>
                     </property>
                    </widget>
                   </item>
                   <item row="2" column="0">
                    <widget class="QLabel" name="circleRadiusLabel">
                     <property name="text">
                      <string>Радиус</string>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="1">
                    <widget class="QSpinBox" name="circleCenterX_SB">
                     <property name="maximum">
                      <number>600</number>
                     </property>
                     <property name="value">
                      <number>200</number>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="2">
                    <widget class="QSpinBox" name="circleCenterY_SB">
                     <property name="maximum">
                      <number>600</number>
                     </property>
                     <property name="value">
                      <number>400</number>
                     </property>
                    </widget>
                   </item>
                   <item row="2" column="1" colspan="2">
                    <widget class="QSpinBox" name="circleRadiusSB">
                     <property name="maximum">
                      <number>600</number>
                     </property>
                     <property name="value">
                      <number>100</number>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <widget class="QPushButton" name="drawCircleBtn">
                   <property name="text">
                    <string>Нарисовать окружность</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QFrame" name="circleSpectrumFrame">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="frameShape">
                 <enum>QFrame::StyledPanel</enum>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_14">
                 <item>
                  <widget class="QLabel" name="circleSpectrumLabel">
                   <property name="text">
                    <string>Параметры спектра</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <layout class="QGridLayout" name="circleSpectrumInputGrid" columnminimumwidth="80,0,0">
                   <item row="2" column="0">
                    <widget class="QLabel" name="circleScpectRbRe_label">
                     <property name="sizePolicy">
                      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                       <horstretch>0</horstretch>
                       <verstretch>0</verstretch>
                      </sizepolicy>
                     </property>
                     <property name="text">
                      <string>(Rнач, Rкон)</string>
                     </property>
                    </widget>
                   </item>
                   <item row="2" column="2">
                    <widget class="QSpinBox" name="circleSpectReSB">
                     <property name="maximum">
                      <number>600</number>
                     </property>
                     <property name="value">
                      <number>220</number>
                     </property>
                    </widget>
                   </item>
                   <item row="2" column="1">
                    <widget class="QSpinBox" name="circleSpectRbSB">
                     <property name="maximum">
                      <number>600</number>
                     </property>
                     <property name="value">
                      <number>10</number>
                     </property>
                    </widget>
                   </item>
                   <item row="4" column="0">
                    <widget class="QLabel" name="circleSpectCountLabel">
                     <property name="text">
                      <string>Количество</string>
                     </property>
                    </widget>
                   </item>
                   <item row="4" column="1" colspan="2">
                    <widget class="QSpinBox" name="circleSpectCountSB">
                     <property name="minimum">
                      <number>1</number>
                     </property>
                     <property name="maximum">
                      <number>999</number>
                     </property>
                     <property name="value">
                      <number>20</number>
                     </property>
                    </widget>
                   </item>
                   <item row="3" column="0">
                    <widget class="QLabel" name="circleSpectStepLabel">
                     <property name="text">
                      <string>Шаг</string>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="0">
                    <widget class="QLabel" name="circleSpectParamsLabel">
                     <property name="sizePolicy">
                      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                       <horstretch>0</horstretch>
                       <verstretch>0</verstretch>
                      </sizepolicy>
                     </property>
                     <property name="text">
                      <string>Параметры</string>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="1" colspan="2">
                    <widget class="QComboBox" name="circleSpectParamsCB">
                     <property name="sizePolicy">
                      <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
                       <horstretch>0</horstretch>
                       <verstretch>0</verstretch>
                      </sizepolicy>
                     </property>
                    </widget>
                   </item>
                   <item row="3" column="1" colspan="2">
                    <widget class="QSpinBox" name="circleSpectStepSB">
                     <property name="minimum">
                      <number>1</number>
                     </property>
                     <property name="maximum">
                      <number>300</number>
                     </property>
                     <property name="value">
                      <number>10</number>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <widget class="QPushButton" name="drawCircleSpectBtn">
                   <property name="text">
                    <string>Нарисовать спектр</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="ellipsePage">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>220</width>
               <height>0</height>
              </size>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_6">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QFrame" name="drawEllipseFrame">
                <property name="frameShape">
                 <enum>QFrame::StyledPanel</enum>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_15">
                 <item>
                  <widget class="QLabel" name="drawEllipseLabel">
                   <property name="text">
                    <string>Параметры эллипса</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <layout class="QGridLayout" name="ellipseInputGrid" columnminimumwidth="80,0,0">
                   <item row="0" column="2">
                    <widget class="QSpinBox" name="ellipseCenterY_SB">
                     <property name="maximum">
                      <number>600</number>
                     </property>
                     <property name="value">
                      <number>200</number>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="0">
                    <widget class="QLabel" name="ellipseCenterLabel">
                     <property name="text">
                      <string>(Xc, Yc)</string>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="1">
                    <widget class="QSpinBox" name="ellipseCenterX_SB">
                     <property name="maximum">
                      <number>600</number>
                     </property>
                     <property name="value">
                      <number>400</number>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="0">
                    <widget class="QLabel" name="ellipseRaRb_label">
                     <property name="text">
                      <string>(Ra, Rb)</string>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="1">
                    <widget class="QSpinBox" name="ellipseRa_SB">
                     <property name="maximum">
                      <number>600</number>
                     </property>
                     <property name="value">
                      <number>130</number>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="2">
                    <widget class="QSpinBox" name="ellipseRb_SB">
                     <property name="maximum">
                      <number>600</number>
                     </property>
                     <property name="value">
                      <number>70</number>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <widget class="QPushButton" name="drawEllipseBtn">
                   <property name="text">
                    <string>Нарисовать эллипс</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QFrame" name="ellipseSpectrumFrame">
                <property name="frameShape">
                 <enum>QFrame::StyledPanel</enum>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_16">
                 <item>
                  <widget class="QLabel" name="ellipseSpectrumLabel">
                   <property name="text">
                    <string>Параметры спектра</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <layout class="QGridLayout" name="ellipseSpectInputGrid" columnminimumwidth="80,0,0">
                   <item row="1" column="1">
                    <widget class="QSpinBox" name="ellipseSpectRaBegin_SB">
                     <property name="maximum">
                      <number>600</number>
                     </property>
                     <property name="value">
                      <number>60</number>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="0">
                    <widget class="QLabel" name="ellipseSpectParamsLabel">
                     <property name="text">
                      <string>Параметры</string>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="1" colspan="2">
                    <widget class="QComboBox" name="ellipseSpectParamsCB"/>
                   </item>
                   <item row="2" column="1" colspan="2">
                    <widget class="QSpinBox" name="ellipseSpectStepSB">
                     <property name="minimum">
                      <number>1</number>
                     </property>
                     <property name="maximum">
                      <number>300</number>
                     </property>
                     <property name="value">
                      <number>10</number>
                     </property>
                    </widget>
                   </item>
                   <item row="2" column="0">
                    <widget class="QLabel" name="ellipseSpectStepLabel">
                     <property name="text">
                      <string>Шаг</string>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="0">
                    <widget class="QLabel" name="ellipseSpectRabBegin_label">
                     <property name="text">
                      <string>(Raн, Rbн)</string>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="2">
                    <widget class="QSpinBox" name="ellipseSpectRbBegin_SB">
                     <property name="maximum">
                      <number>600</number>
                     </property>
                     <property name="value">
                      <number>40</number>
                     </property>
                    </widget>
                   </item>
                   <item row="3" column="0">
                    <widget class="QLabel" name="ellipseSpectCountLabel">
                     <property name="text">
                      <string>Количество</string>
                     </property>
                    </widget>
                   </item>
                   <item row="3" column="1" colspan="2">
                    <widget class="QSpinBox" name="ellipseSpectCountSB">
                     <property name="minimum">
                      <number>1</number>
                     </property>
                     <property name="maximum">
                      <number>999</number>
                     </property>
                     <property name="value">
                      <number>10</number>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <widget class="QPushButton" name="drawEllipseSpectrumBtn">
                   <property name="text">
                    <string>Нарисовать спектр</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="bottomLayout">
          <property name="spacing">
           <number>4</number>
          </property>
          <item>
           <widget class="QPushButton" name="timeCompButton">
            <property name="text">
             <string>Сравнить время</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="cleanScreenButton">
            <property name="text">
             <string>Очистить экран</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QFrame" name="graphicsViewFrame">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QGraphicsView" name="graphicsView"/>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>950</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="help">
    <property name="title">
     <string>справка</string>
    </property>
    <addaction name="aboutTaskAction"/>
   </widget>
   <addaction name="help"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="fileImportAction">
   <property name="text">
    <string>импорт модели</string>
   </property>
  </action>
  <action name="aboutTaskAction">
   <property name="text">
    <string>О программе</string>
   </property>
  </action>
  <action name="fileExportAction">
   <property name="text">
    <string>экспорт модели</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
